source/opt/set_spec_constant_default_value_pass.cpp

```diff
@@ -251,16 +251,15 @@ bool SetSpecConstantDefaultValuePass::Process(ir::Module* module) {
 }
 
 // Returns true if the char pointed by the given pointer is '\0' or '}'.
-bool IsEnd(const char* const ptr) {
-  if (*ptr == '\0' || *ptr == '}') {
-    return true;
-  }
-  return false;
+inline bool IsEnd(const char* const ptr) {
+  if (!ptr) return false;
+  return *ptr == '\0' || *ptr == '}';
 }
 
 // Returns true if the char pointed by the given pointer is ' ', '\n', '\t',
 // '\r'.
-bool IsSpace(const char* const ptr) {
+inline bool IsSpace(const char* const ptr) {
+  if (!ptr) return false;
   if (*ptr == ' ' || *ptr == '\n' || *ptr == '\r' || *ptr == '\t') {
```

Comment:
This can also be written as `return *ptr == ' ' || ...`, which is less verbose, and it's also quite readable.

Same for the following function.